docs/tech/rfc/0019-240310-cesium-race-conditions.md

```diff
@@ -1,4 +1,4 @@
-# 1 - Cesium - Channel Segment Storage Engine
+# 19 - Cesium Race Conditions
```

Comment:
Overall, this is much better. I thought you covered all of the ground you needed to and made clear you systemically looked for all locations where race conditions could be. The explanation of Cesium is much clearer. I mostly wrote comments on stylistic and grammatical changes so we could focus on getting this looking better and merged (this PR has been open for a month). I do have a few points of major feedback:
- The job of technical writing is to convey information to the reader. The first step of this is putting all the information in the document, which anybody can do fine. The next step is much harder: making it easy for the reader to understand what you are saying. A few suggestions for how to do that:
  - Use document structure to your advantage. The document structure should be clean and logical. Try to avoid anything past `<h3>`, and use lists instead to prevent what would be sub-sub-subsection (`<h4>`) and more.
  - Use things like lists, tables, and images to make information more easily digestible.
  - Avoid complex sentence structure. Try to get rid of long run-ons, colons, semicolons, and dashes. These usually can just be several shorter sentences.
  - After you have written a rough draft, go line-by-line and try deleting a sentence if it is not useful.
- Get rid of weird capitalizations. There are a lot of "Garbage Collection" and "Unary" and other words that should be lowercase.
- Avoid acronyms. Even though "DB" is a common acronym seen in programming, it is almost always used in the context of variable names or database products like CockroachDB and MongoDB. I think the writing looks more professional and reads cleaner when you write out "database".
- Use more code variables! There are plenty of variable names in here that should be denoted with backticks to make it clear you are referring to a variable.
- Have a clear capitalization scheme for section titles. Emiliano and I have been doing titlecase for `<h1>`-`<h3>` and sentence case for all others.

Reply:
Thanks very helpful. I included more code variables and avoided DB where possible – there are still lots of small titles, but I felt like that might too big of an overhaul at this stage. I also fixed casing. Will remember the rest for the next rfc.